src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java

```diff
@@ -1,50 +1,116 @@
 package com.minecolonies.coremod.blocks;
 
 import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraftforge.fml.common.registry.IForgeRegistry;
 
 /**
  * Class to create the modBlocks.
  * References to the blocks can be made here
+ *
+ * We disabled the following finals since we are neither able to mark the items as final, nor do we want to provide public accessors.
  */
-public final class ModBlocks
+@SuppressWarnings({"squid:ClassVariableVisibilityCheck", "squid:S2444", "squid:S1444"})
+public class ModBlocks
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this class so it has no more than 20 public fields, rather than the 25 it currently has. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1820)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this class "final" or add a public constructor. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2974)